File: JavaScript/6kyu/Sum The Tree.js
// return the sum of all values in the tree, including the root
function sumTheTreeValues(root){
  if(root.left==null && root.right==null) return root.value;
  if(root.left!=null && root.right==null) return root.value+sumTheTreeValues(root.left);
  if(root.left!=null && root.right!=null) return root.value+sumTheTreeValues(root.left)+sumTheTreeValues(root.right);
  if(root.left==null && root.right!=null) return root.value+sumTheTreeValues(root.right);

}

File: C#/6kyu/Sum The Tree.cs
using System;

public static class Kata
{
  public static int SumTree(Node root)
  {

    if(root.Left==null && root.Right==null) return root.Value;
    if(root.Left!=null && root.Right==null) return root.Value+SumTree(root.Left);
    if(root.Left!=null && root.Right!=null) return root.Value+SumTree(root.Left)+SumTree(root.Right);
    if(root.Left==null && root.Right!=null) return root.Value+SumTree(root.Right);
  return 0;}
}
